legacy-fabric-keybindings-api-v1/1.6.4/src/main/java/net/legacyfabric/fabric/impl/client/keybinding/OptimizedListWidget.java

```diff
@@ -0,0 +1,396 @@
+/*
+ * Copyright (c) 2020 - 2024 Legacy Fabric
+ * Copyright (c) 2016 - 2022 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.legacyfabric.fabric.impl.client.keybinding;
+
+import org.lwjgl.input.Mouse;
+import org.lwjgl.opengl.GL11;
+
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.gui.DrawableHelper;
+import net.minecraft.client.gui.widget.ButtonWidget;
+import net.minecraft.client.render.Tessellator;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+
+@Environment(EnvType.CLIENT)
+public abstract class OptimizedListWidget {
+	private final MinecraftClient client;
+	private int field_1247;
+	private int height;
+	protected int yStart;
+	protected int yEnd;
+	private int xEnd;
+	private int field_1250;
+	protected final int entryHeight;
+	private int homeButtonId;
+	private int endButtonId;
+	protected int lastMouseX;
+	protected int lastMouseY;
+	private float field_1253 = -2.0F;
+	private float scrollSpeed;
+	private float scrollAmount;
+	private int selectedEntry = -1;
+	private long time;
+	private boolean field_1258 = true;
```

Comment:
What are these names

Reply:
This class is a modified version of vanilla ListWidget with fixes related to keyboard and mouse inputs detection. I haven't renamed missing mappings names yet.